### PR TITLE
[FIX] allow only livechat-agents to be contact manager for any omnichannel contact 

### PR DESCRIPTION
**Pull request by murtaza98**
## Issue(s)

image

## Further comments

Tasks:

- [x] Frontend change on auto-select component
- [x] Backend API checks

### CODE REVIEW

**Review by KevLehman — COMMENTED** [review]
What if some random customer has a non-agent as contact manager? 👀

**Comment by murtaza98** [comment]
> What if some random customer has a non-agent as contact manager? :eyes:

But that's not the way this feature was designed for. The whole idea of the Contact manager is that if that manager is online and the visitor tries to contact the company, then the manager the chat reaches directly to this manager instead of any other agent. So if a non-agent is assigned as a contact manager to any contact, then they won't be able to take the chat (since only agents can serve a chat) - hence I believe we need to add this filter over here to prevent such cases

**Comment by KevLehman** [comment]
No no, that's fine. I mean, what happens if some customer _already_ has non-agents as contact managers after this fix is merged 👀 
Will UI crash? Will something odd happen? 😬

**Comment by murtaza98** [comment]
> No no, that's fine. I mean, what happens if some customer _already_ has non-agents as contact managers after this fix is merged eyes Will UI crash? Will something odd happen? grimacing

@KevLehman I checked this case & I didn't find anything odd. Nothing breaks on the UI, however the only thing is that the existing contact manager (who isn't a agent) won't appear in the select dropdown

**Review by KevLehman — DISMISSED** [review]

**Comment by murtaza98** [comment]
Making this a draft for now since blocked on agent's page not getting loaded

PS: suggestions from the QA team to handle a case for a user losing an agent role.

**Comment by murtaza98** [comment]
Hi @KevLehman @RocketChat/chat-engine May I ask for your approvals again? I lost earlier approvals due to some changes which QA team suggested

File: apps/meteor/app/livechat/server/lib/Contacts.js
Comment (KevLehman) [inline apps/meteor/app/livechat/server/lib/Contacts.js:20]: Personally i don't like checking for roles here, but for now that would work.

Pls create a task for changing this check to be a permission in the future 👀

Comment (murtaza98) [reply apps/meteor/app/livechat/server/lib/Contacts.js:20]: I agree to tackle this issue separately since we also have similar usage elsewhere in the code like https://github.com/RocketChat/Rocket.Chat/blob/993745d3996450036e8c39150a0b6926cdbbd869/apps/meteor/app/livechat/server/api/lib/users.js#L51

Here's the task for it btw: https://app.clickup.com/t/2f82np6

**Review by ggazzo — APPROVED** [review]

**Pull request merged**
